src/flows/components/header/Submit.tsx

```diff
@@ -60,11 +60,21 @@ export const Submit: FC = () => {
       .length
   }, [flow.data])
 
+  // Returns a string to avoid array deep comparison woes
+  // Triggers a re-run any time any aggregate function is changed
+  const aggregateOfAggregates = useMemo(() => {
```

Comment:
Would love to hear @drdelambre 's opinion on this

Reply:
so a notebook is a collection of pipes, and to keep everything loosely coupled, we need to ensure that one pipe type has no concept of the internal logic of any other pipe type. as an antithesis: if Pipe A relies on Pipe B having a function in it that transforms a string into a number, then anytime we change Pipe B, we must also ensure that Pipe A still works, and we wont be able to delete Pipe B without rewriting Pipe A. within notebooks, a pipe shouldn't even know that there exists a list of pipes somewhere, it should think of itself as an island at the center of a black hole of loneliness. The notebook itself should know that there is a list of pipes, but shouldn't expect there to be a difference between the implementation of Pipe A vs Pipe B vs... they're all just pipes.

A good rule of thumb for testing coupling is that, in a pure absolutist fantasy, you should be able to delete any module without having to change anything. just go nuclear, delete the file, and the app should keep running just fine.

this is the ideal to chase, but usually some concessions must be made.

the only sore point i might see in this one is that it forces the notebook to know that there is a list of pipes, that there is a pipe of type `metricSelector` (all pipes have the property `type`, but we shouldn't expect any particular value to exist in that space), and that a pipe of type metricSelector implements a property of `agregateFunction`. To decouple this, i would probably flip the filter an map to look like:
```
flow.data.all
  .map(p => p.aggregateFunction?.name)
  .filter(p => !!p)
  .join('')
```

then it doesn't expect any implementation to exist from a pipe, but instead uses any aggregate function that is defined.